e2e/browser-runner/stencil.test.tsx

```diff
@@ -96,4 +96,61 @@ describe('Stencil Component Testing', () => {
         page.unmount()
         await expect(page.root).not.toBeExisting()
     })
+
+    it('can auto peirce shadow dom', async () => {
+        render({
+            components: [AppProfile, NestedComponent],
+            template: () => (
+                <div>
+                    {/* @ts-ignore: types don't exist as we don't compile the components with Stencil */}
+                    <nested-component id="first transparent component"></nested-component>
+                    <div className="nested">
+                        {/* @ts-ignore: types don't exist as we don't compile the components with Stencil */}
+                        <nested-component id="transparent component in a nested context"></nested-component>
+                    </div>
+                    <div className="nested second">
+                        {/* @ts-ignore: types don't exist as we don't compile the components with Stencil */}
+                        <nested-component id="transparent component in a second nested context"></nested-component>
```

Comment:
I think it would be good to have a testcase for a nested-component inside a nested-component 👍

Reply:
I just saw you added much more tests further down so this might not be needed

Reply:
Jup, the `app-profile` component has multiple levels of nested shadow doms and covers this.